### ISSUE TITLE
Getting image and depth bounding boxes

### ISSUE THREAD

**Issue by hawesie** [strands-project/strands_perception_people#116]
I have a student who is working with the tracker. For each observation of a person we'd like to get the part of the RGBD image which contains them, i.e. the actual bounding box or the bounding cylinder which is rendered in the output image of the tracker. Is this possible? We looked at the outputs available and didn't see anything obvious.

**Comment by cdondrup** [comment]
The depth bounding box is published as `/upper_body_detector/detections` as [this message type](https://github.com/strands-project/strands_perception_people/blob/hydro-devel/upper_body_detector/msg/UpperBodyDetector.msg) where everything but `dist` describes an attribute of the box. `dist` is just the distance to the template and therefore reflects the certainty where as `median_depth` represents the `z` value.

regarding the rgb boundingbox, this is currently not published but is just taken from the above message:

```
int x =(int) detections.pos_x[i];
int y =(int) detections.pos_y[i];
int w =(int) detections.width[i];
int h =(int) detections.height[i];
```

and then drawn on the rgb image. This assumes that the images are rectified.

**Comment by hawesie** [comment]
Cool, thanks.

As this is just upper body, how about the complete box around the human? I guess we could get the ground plane (this is standalone, so we'd need to get the extracted one) and project the upper body box downwards, but it looks like this data is in the pedestrian tracker already.

**Comment by cdondrup** [comment]
Afaik, the tracker only renders a bounding cylinder onto the RGB image and does not export it anywhere. See [this line](https://github.com/strands-project/strands_perception_people/blob/hydro-devel/mdl_people_tracker/src/Tracker.cpp#L403) for where it calls the rendering function. Maybe that gives you an idea on how to do it.

I know too little about the tracker to make a more qualified statement I'm afraid. Maybe @lucasb-eyer knows or knows someone who knows.

**Comment by cdondrup** [comment]
Regarding the ground plane, that is published as `/ground_plane` and should have the same timestamp as the upper_body_detector because they both get their timestamps from the image they use. Can therefore easily be synchronised.

**Comment by lucasb-eyer** [comment]
I know less of the tracker than you (and I hope that it'll stay that way :grinning:) but I asked a colleague who is starting to get into it. The position of the person should be published already(?) For the visualization, the cylinders have a hardcoded width/radius and for the height, it's taken from the detection, but @cdondrup's link looks more useful than anything I'd be able to point to.

**Comment by cdondrup** [comment]
Indeed, the position of the human in world and camera coordinates projected to the ground plane is already [published](https://github.com/strands-project/strands_perception_people/blob/hydro-devel/mdl_people_tracker/msg/MdlPeopleTracker.msg). Maybe that helps as well @hawesie because that is already using the estimated ground plane.

**Comment by cdondrup** [comment]
Is this still valid? @hawesie

**Comment by hawesie** [comment]
We have a solution working, but nothing generalised enough for public consumption.

**Comment by lucasb-eyer** [comment]
Reopened since @vonovak is currently working on this.

**Comment by lucasb-eyer** [comment]
cf. #148
